src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseService.java

```diff
@@ -65,91 +195,411 @@
+      String errorMsg =
+          String.format(
+              WRONG_OLD_SAMPLE_UNIT_TYPE_MSG,
+              oldCaseSampleUnitType,
+              expectedOldCaseSampleUnitTypes);
+      log.error(errorMsg);
+      throw new CTPException(CTPException.Fault.VALIDATION_FAILED, errorMsg);
+    }
+  }
 
   /**
-   * Adds a new row to the caseiacaudit table with the case id and iac of given case
+   * To compare the old case sample unit type with the expected sample unit types
    *
-   * @param updatedCase the CaseCreation data
+   * @param oldCaseSampleUnitType the old case sample unit type
+   * @param expectedOldCaseSampleUnitTypes a comma separated list of expected sample unit types
+   * @return true if the expected types contain the old case sample unit type
    */
-  void saveCaseIacAudit(Case updatedCase);
+  private boolean compareOldCaseSampleUnitType(
+      String oldCaseSampleUnitType, String expectedOldCaseSampleUnitTypes) {
+    boolean result = false;
+    if (expectedOldCaseSampleUnitTypes != null) {
+      List<String> expectedTypes = Arrays.asList(expectedOldCaseSampleUnitTypes.split("\\s*,\\s*"));
+      if (oldCaseSampleUnitType != null && expectedTypes.contains(oldCaseSampleUnitType)) {
+        result = true;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Create the new Case.
+   *
+   * @param caseData SampleUnit from which to create Case.
+   * @param caseGroup to which Case belongs.
+   * @return newCase created Case.
+   */
+  private Case createNewCase(SampleUnit caseData, CaseGroup caseGroup) {
+    Case newCase = new Case();
+    newCase.setId(UUID.randomUUID());
+
+    // values from case group
+    newCase.setCaseGroupId(caseGroup.getId());
+    newCase.setCaseGroupFK(caseGroup.getCaseGroupPK());
+    newCase.setSampleUnitId(UUID.fromString(caseData.getId()));
+
+    // set case values from sampleUnit
+    newCase.setSampleUnitType(SampleUnitDTO.SampleUnitType.valueOf(caseData.getSampleUnitType()));
+
+    if (caseData.getPartyId() != null) {
+      newCase.setPartyId(UUID.fromString(caseData.getPartyId()));
+    }
+
+    newCase.setCollectionInstrumentId(UUID.fromString(caseData.getCollectionInstrumentId()));
+    newCase.setActionPlanId(UUID.fromString(caseData.getActionPlanId()));
+
+    // HardCoded values
+    newCase.setState(CaseState.SAMPLED_INIT);
+    newCase.setCreatedDateTime(DateTimeUtil.nowUTC());
+    newCase.setCreatedBy(Constants.SYSTEM);
+
+    return newCase;
+  }
+
+  /**
+   * Check to see if the event requires a response to be recorded for the case and if so ... record
+   * it
+   *
+   * @param category the category details of the event
+   * @param targetCase the 'source' case the event is being created for
+   * @param timestamp timestamp the timestamp of the CaseResponse
+   */
+  private void recordCaseResponse(Category category, Case targetCase, Timestamp timestamp) {
+    InboundChannel channel = null;
+    switch (category.getCategoryName()) {
```

Comment:
interestingly this 13-line switch statement can be rewritten as 7 lines of `if..else if...else if`. I hate switch statements